src/map/grid.rs

```diff
@@ -15,3 +15,18 @@ pub struct Grid {
     pub rows: usize,
     pub tiles: Vec<Arc<Tile>>,
 }
+
+impl Grid {
+    pub fn passable(&self, tilepos: (isize, isize)) -> bool {
+        if tilepos.0 < 0 || tilepos.1 < 0 {
+            return false;
+        }
+        let p0 = tilepos.0 as usize;
+        let p1 = tilepos.1 as usize;
```

Comment:
we are already converting to usize, so we could do an unsigned comppare against the collum length, in which case comparing to zero would not be necessary

Reply:
@jesings ^